### DISCUSSION TITLE
使用 defineConfig 配置路由不符合预期

### DISCUSSION THREAD

**Discussion by changhuaixin** [umijs/dumi#2166] [Q&A]
1. 在 .dumirc.ts 里通过 defineConfig 定义 3 个路由
  routes: [
    {
      path: '/framescope',
      component: '../../../framescope/src/layouts/index',
    },
    {
      path: '/framescope/hotspot',
      component: '../../../framescope/src/pages/Hotspots/index',
    },
    {
      path: '/framescope/service',
      component: '../../../framescope/src/pages/uploadFile/index',
    },
2. 导航栏显示第 1 个路由，符合预期
![image](https://github.com/umijs/dumi/assets/61184708/9b5dde6f-533d-4ea4-8a46-7a26d2efd411)

3. 点击跳转到第 2 条路由，期待跳转到第 1 条
![image](https://github.com/umijs/dumi/assets/61184708/212eb8d6-f740-41f7-84b7-de07edbd10cc)

期待跳转到的链接是 http://localhost:8000/framescope
实际是 http://localhost:8000/framescope/hotspot

**Comment by changhuaixin** [comment]
感觉当前的规则是根据顶层路由生成一级导航，导航栏内容默认选择排在第 1 位的页面。
path: '/framescope',
path: '/framescope/hotspot',
path: '/framescope/service',
处理成 1 个一级导航页面

实测使用 order: 1 可以控制页面顺序，在对应的 react 组件里写
/**
 * nav: layouts
 * sidebar: false
 * order: 1
 */
就可以让他作为一级导航的链接内容。

这是 by-design 的行为吗

**Reply by PeachScript** [reply]
不是 by-design 的，配置式路由的用法基本没有测试过，所以可能是存在 bug 的，欢迎 PR 帮忙完善
